### ISSUE TITLE
No login screen after new install

### ISSUE THREAD

**Issue by dfdario** [Sandmann79/xbmc#326]
After last Widevine update days ago I wasn't able to watch any movie because of only audio and no video was available. Then I decided to start from scratch by deleting ~/.kodi directory.
I installed only the Amazon VOD and anything else then I tried to perform login.
Issue is: no screen is presented but only the rolling wheel.
I would like to point out that until Widevine update Amazon VOD was working perfectly while after that all other plugins are still working (Netflix, RAI replay, La7 replay, etc) but Amazon VOD.
Systems where I am testing on: Debian buster / Debian stretch - Kodi 18.4 / 18.3 (from Debian multimedia repo)
log reports:
  DEBUG: ------ Window Deinit (DialogAddonSettings.xml) ------                             │
│2019-09-17 18:09:34.110 T:140643027732224   ERROR: [Amazon VOD] Error reason: HTTPSConnectionPool(host='na.api.amazonvideo.com', port│
│2019-09-17 18:09:34.110 T:140643027732224    INFO: CPythonInvoker(2, /home/cicero/.kodi/addons/plugin.video.amazon-test/default.py)│
│2019-09-17 18:09:36.595 T:140644758285504   DEBUG: ------ Window Deinit (Pointer.xml) ------  
Further info: Plugin works properly on Mi-box S  and  Kodi Android (due to different Android decryption)

More log after purge kodi, delete /usr/share/kodi and ~/kodi and reinstall.
2019-09-17 19:38:19.597 T:140663072230592  NOTICE: ADDON: xbmc.addon v18.4.0 installed
2019-09-17 19:38:19.597 T:140663072230592  NOTICE: ADDON: xbmc.core v0.1.0 installed
2019-09-17 19:38:19.597 T:140663072230592  NOTICE: ADDON: xbmc.gui v5.14.0 installed
2019-09-17 19:38:19.597 T:140663072230592  NOTICE: ADDON: xbmc.json v10.3.0 installed
2019-09-17 19:38:19.597 T:140663072230592  NOTICE: ADDON: xbmc.metadata v2.1.0 installed
2019-09-17 19:38:19.597 T:140663072230592  NOTICE: ADDON: xbmc.python v2.26.0 installed
2019-09-17 19:38:19.597 T:140663072230592  NOTICE: ADDON: xbmc.webinterface v1.0.0 installed
2019-09-17 19:38:19.659 T:140663072230592  NOTICE:   load skin from: /usr/share/kodi/addons/skin.estuary (version: 2.0.25)
2019-09-17 19:38:57.164 T:140661244290816  NOTICE: [Amazon VOD] Switching UserAgent
2019-09-17 19:38:57.165 T:140661244290816  NOTICE: Previous line repeats 1 times.
2019-09-17 19:38:57.165 T:140661244290816  NOTICE: [Amazon VOD] Loading list of common UserAgents
2019-09-17 19:38:57.167 T:140661244290816  NOTICE: [Amazon VOD] getURL: https://techblog.willshouse.com/2012/01/03/most-common-user-agents/
2019-09-17 19:38:57.167 T:140661219211008  NOTICE: [Amazon VOD] Loading list of common UserAgents
2019-09-17 19:38:57.170 T:140661219211008  NOTICE: [Amazon VOD] getURL: https://techblog.willshouse.com/2012/01/03/most-common-user-agents/
2019-09-17 19:39:01.077 T:140661219211008   ERROR: [Amazon VOD] Error reason: 503 error (TryAgain)
2019-09-17 19:39:01.078 T:140661219211008  NOTICE: [Amazon VOD] Attempt #2
2019-09-17 19:39:01.078 T:140661219211008  NOTICE: [Amazon VOD] getURL: https://techblog.willshouse.com/2012/01/03/most-common-user-agents/
2019-09-17 19:39:01.103 T:140661244290816   ERROR: [Amazon VOD] Error reason: 503 error (TryAgain)
2019-09-17 19:39:01.103 T:140661244290816  NOTICE: [Amazon VOD] Attempt #2
2019-09-17 19:39:01.103 T:140661244290816  NOTICE: [Amazon VOD] getURL: https://techblog.willshouse.com/2012/01/03/most-common-user-agents/
2019-09-17 19:39:05.076 T:140661244290816   ERROR: [Amazon VOD] Error reason: 503 error (TryAgain)
2019-09-17 19:39:05.077 T:140661244290816  NOTICE: [Amazon VOD] Attempt #3
2019-09-17 19:39:05.077 T:140661244290816  NOTICE: [Amazon VOD] getURL: https://techblog.willshouse.com/2012/01/03/most-common-user-agents/
2019-09-17 19:39:05.085 T:140661219211008   ERROR: [Amazon VOD] Error reason: 503 error (TryAgain)
2019-09-17 19:39:05.085 T:140661219211008  NOTICE: [Amazon VOD] Attempt #3
2019-09-17 19:39:05.085 T:140661219211008  NOTICE: [Amazon VOD] getURL: https://techblog.willshouse.com/2012/01/03/most-common-user-agents/
2019-09-17 19:39:09.082 T:140661219211008   ERROR: [Amazon VOD] Error reason: 503 error (TryAgain)
2019-09-17 19:39:09.083 T:140661219211008  NOTICE: [Amazon VOD] Using UserAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
2019-09-17 19:39:09.126 T:140661244290816   ERROR: [Amazon VOD] Error reason: 503 error (TryAgain)
2019-09-17 19:39:09.133 T:140661219211008 WARNING: CPythonInvoker(2, /home/cicero/.kodi/addons/plugin.video.amazon-test/default.py): the python script "/home/cicero/.kodi/addons/plugin.video.amazon-test/default.py" has left several classes in memory that we couldn't clean up. The classes include: N9XBMCAddon9xbmcaddon5AddonE,N9XBMCAddon4xbmc7MonitorE,N9XBMCAddon7xbmcgui6DialogE
2019-09-17 19:39:09.133 T:140661244290816  NOTICE: [Amazon VOD] Using UserAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
2019-09-17 19:39:09.133 T:140661244290816  NOTICE: [Amazon VOD] Retrieve territoral config
2019-09-17 19:39:09.134 T:140661244290816  NOTICE: [Amazon VOD] getURL: https://na.api.amazonvideo.com/cdp/usage/v2/GetAppStartupConfig?deviceTypeID=A28RQHJKHM2A2W&deviceID=148c5202492a035c2d008b17af2dabe1454ede2506f8398f75b93497&firmware=1&version=1&format=json
2019-09-17 19:39:09.146 T:140661252683520   ERROR: GetDirectory - Error getting plugin://plugin.video.amazon-test/
2019-09-17 19:39:10.745 T:140661244290816   ERROR: [Amazon VOD] Error reason: HTTPSConnectionPool(host='na.api.amazonvideo.com', port=443): Max retries exceeded with url: /cdp/usage/v2/GetAppStartupConfig?deviceTypeID=A28RQHJKHM2A2W&deviceID=148c5202492a035c2d008b17af2dabe1454ede2506f8398f75b93497&firmware=1&version=1&format=json (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl_choose_client_version', 'unsupported protocol')],)",),)) (SSLError)
2019-09-17 19:39:10.786 T:140661244290816 WARNING: CPythonInvoker(3, /home/cicero/.kodi/addons/plugin.video.amazon-test/default.py): the python script "/home/cicero/.kodi/addons/plugin.video.amazon-test/default.py" has left several classes in memory that we couldn't clean up. The classes include: N9XBMCAddon4xbmc7MonitorE,N9XBMCAddon7xbmcgui6DialogE,N9XBMCAddon9xbmcaddon5AddonE
2019-09-17 19:39:20.316 T:140663072230592   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.amazon-test/) failed
2019-09-17 19:39:20.926 T:140662024419072  NOTICE: [Amazon VOD] Retrieve territoral config
2019-09-17 19:39:20.927 T:140662024419072  NOTICE: [Amazon VOD] getURL: https://na.api.amazonvideo.com/cdp/usage/v2/GetAppStartupConfig?deviceTypeID=A28RQHJKHM2A2W&deviceID=148c5202492a035c2d008b17af2dabe1454ede2506f8398f75b93497&firmware=1&version=1&format=json
2019-09-17 19:39:21.316 T:140662024419072   ERROR: [Amazon VOD] Error reason: HTTPSConnectionPool(host='na.api.amazonvideo.com', port=443): Max retries exceeded with url: /cdp/usage/v2/GetAppStartupConfig?deviceTypeID=A28RQHJKHM2A2W&deviceID=148c5202492a035c2d008b17af2dabe1454ede2506f8398f75b93497&firmware=1&version=1&format=json (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl_choose_client_version', 'unsupported protocol')],)",),)) (SSLError)
2019-09-17 19:39:21.361 T:140662024419072 WARNING: CPythonInvoker(4, /home/cicero/.kodi/addons/plugin.video.amazon-test/default.py): the python script "/home/cicero/.kodi/addons/plugin.video.amazon-test/default.py" has left several classes in memory that we couldn't clean up. The classes include: N9XBMCAddon9xbmcaddon5AddonE,N9XBMCAddon4xbmc7MonitorE,N9XBMCAddon7xbmcgui6DialogE
2019-09-17 19:39:40.189 T:140662385145600  NOTICE: [Amazon VOD] Retrieve territoral config
2019-09-17 19:39:40.191 T:140662385145600  NOTICE: [Amazon VOD] getURL: https://na.api.amazonvideo.com/cdp/usage/v2/GetAppStartupConfig?deviceTypeID=A28RQHJKHM2A2W&deviceID=148c5202492a035c2d008b17af2dabe1454ede2506f8398f75b93497&firmware=1&version=1&format=json
2019-09-17 19:39:40.490 T:140662385145600   ERROR: [Amazon VOD] Error reason: HTTPSConnectionPool(host='na.api.amazonvideo.com', port=443): Max retries exceeded with url: /cdp/usage/v2/GetAppStartupConfig?deviceTypeID=A28RQHJKHM2A2W&deviceID=148c5202492a035c2d008b17af2dabe1454ede2506f8398f75b93497&firmware=1&version=1&format=json (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl_choose_client_version', 'unsupported protocol')],)",),)) (SSLError)
2019-09-17 19:40:52.554 T:140663047026432  NOTICE: PulseAudio: Opened device Default in pcm mode with Buffersize 150 ms
2019-09-17 19:40:52.578 T:140662385145600 WARNING: CPythonInvoker(5, /home/cicero/.kodi/addons/plugin.video.amazon-test/default.py): the python script "/home/cicero/.kodi/addons/plugin.video.amazon-test/default.py" has left several classes in memory that we couldn't clean up. The classes include: N9XBMCAddon7xbmcgui6DialogE,N9XBMCAddon9xbmcaddon5AddonE,N9XBMCAddon4xbmc7MonitorE
2019-09-17 19:40:52.578 T:140661638551296   ERROR: GetDirectory - Error getting plugin://plugin.video.amazon-test/
2019-09-17 19:40:52.582 T:140663072230592   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.amazon-test/) failed
2019-09-17 19:40:57.960 T:140663072230592  NOTICE: Stopping player
2019-09-17 19:40:57.960 T:140663072230592  NOTICE: Storing total System Uptime
2019-09-17 19:40:57.960 T:140663072230592  NOTICE: Saving settings
2019-09-17 19:40:57.962 T:140663072230592  NOTICE: Saving skin settings
2019-09-17 19:40:57.963 T:140663072230592  NOTICE: Stopping all
2019-09-17 19:40:57.963 T:140663072230592  NOTICE: ES: Stopping event server
2019-09-17 19:40:57.963 T:140663072230592  NOTICE: stopping zeroconf publishing
2019-09-17 19:40:58.356 T:140661655336704  NOTICE: ES: UDP Event server stopped
2019-09-17 19:40:58.360 T:140663072230592  NOTICE: stop dvd detect media
2019-09-17 19:40:58.789 T:140661227603712  NOTICE: [Amazon VOD] Service: Proxy server stopped
2019-09-17 19:40:58.789 T:140661227603712  NOTICE: [Amazon VOD] Service stopped
2019-09-17 19:40:58.794 T:140661227603712 WARNING: CPythonInvoker(1, /home/cicero/.kodi/addons/plugin.video.amazon-test/service.py): the python script "/home/cicero/.kodi/addons/plugin.video.amazon-test/service.py" has left several classes in memory that we couldn't clean up. The classes include: N9XBMCAddon9xbmcaddon5AddonE,N9XBMCAddon4xbmc7MonitorE,N9XBMCAddon7xbmcgui6DialogE
2019-09-17 19:40:59.015 T:140663072230592  NOTICE: Application stopped
2019-09-17 19:40:59.216 T:140663072230592  NOTICE: XBApplicationEx: destroying...
2019-09-17 19:40:59.583 T:140663072230592  NOTICE: unload skin
2019-09-17 19:40:59.605 T:140663072230592  NOTICE: unload sections
2019-09-17 19:40:59.637 T:140663072230592  NOTICE: XBApplicationEx: application stopped!

**Comment by Varstahl** [comment]
Well, I haven't developed nor find the time to use the addon myself recently, but by looking at what you just posted, it seems that amazon not only did more changes to the backend but straight up changed endpoints. Unless it's some random offshoot error, it seems that the Widevine change brought changes on infrastructures as well.

I can't tell what that change entails as I'm still working hard to meet a deadline for a client, but it's going to be interesting…

**Comment by dfdario** [comment]
Before what I mentioned in the bug report, after Widevine was updated
the first time I was able to watch Android video again by applying a
workaround.Stating that new libwidevinecdm.so was installed in
~/.kodi/cdm/libwidevinecdm.so and the old one in
~/.kodi/userdata/addon_data/script.module.inputstreamhelper/cdm/libwide
vinecdm.soI deleted the newer in  ~/.kodi/cdm/ and linked the oldone by
this "ln -s
~/.kodi/userdata/addon_data/script.module.inputstreamhelper/cdm/libwide
vinecdm.so ~/.kodi/cdm/"Now this does not solve anymore.
Cheers

Il giorno mar, 17/09/2019 alle 10.20 -0700, Bruno Passeri ha scritto:

**Comment by dfdario** [comment]
It seems that host name passed is not more valid
https:///cdp/usage/v2/GetAppStartupConfig?deviceTypeID=A28RQHJKHM2A2W: Nome dell'host non valido.

**Comment by Varstahl** [comment]
I need to look into it personally I think, the change goes deeper than simply a slight UI change, they changed their infrastructure and possibly most of their endpoints. It's going to take me at least 2-3 days until I can get my hands on the issue though.

**Comment by dfdario** [comment]
OK I have found the trick.
To get in I have to choose manually the region ROE Europe. I am guessing the problem in logging in reside in the "Auto" Region selection.
But now I have another problem: by playing I have only sound and no video. The video remains to the movies list page.
Since this is due to the libwidevinecdm.so in the past I solved it by deleting .kodi/cdm/libwidevinecdm.so and link it to .kodi/userdata/addon_data/script.module.inputstreamhelper/cdm/libwidevinecdm.so but now this trick doesn't work anymore.

**Comment by Varstahl** [comment]
Sorry, for the late answer. There was/is a problem with Kodi, where it would fire requests multiple times, resulting in the login prompt triggering in multiples of two. Due to the way the framework works, the second call would hide the first, making it impossible to enter the credentials. This is fixed in 17ae68a2bc105121a039f68bc0582cb18c1a5cc9 but not yet live, as some more work and multiple tests are needed.

**Comment by Goommer** [comment]
@dfdario You use debian and the error is ssl, may be the problem is the use of TLS1.0 (na.api.amazonvideo.com) and the restriction of Debian to use at least TLS1.2
Under your own risk, you can edit the /etc/ssl/openssl.cnf file commenting or modifying the last two lines:
```
[system_default_sect]
MinProtocol = TLSv1.2
CipherString = DEFAULT@SECLEVEL=2
```
I had the same problem and I solved it that way, but I still can't login
